File: 3005.count-elements-with-maximum-frequency/solution.kt
class Solution {
    fun maxFrequencyElements(nums: IntArray): Int {
        val histogram = IntArray(101)
        var maxFrequency = 0
        var result = 0
        nums.forEach { n ->
            histogram[n]++
            if (histogram[n] == maxFrequency) {
                result += maxFrequency
            } else if (histogram[n] > maxFrequency) {
                maxFrequency = histogram[n]
                result = maxFrequency
            }
        }
        return result
    }
}

File: 3005.count-elements-with-maximum-frequency/solution.go
func maxFrequencyElements(nums []int) int {
    var f [101]int
    var max, res int
    
    for _, n := range nums {
        f[n]++
        if f[n] == max {
            res += max
        } else if f[n] > max { 
            max = f[n]
            res = max
        }
    } 
    
    return res
}